WeakAuras/RegionTypes/AuraBar.lua

```diff
@@ -277,6 +315,11 @@ local barPrototype = {
       self.fgFrame:SetHeight(yProgress > 0.0001 and yProgress or 0.0001);
     end
 
+    -- Apply updated gradient
+    if self.updateGradient then
+      self:GetParent():Color(self:GetParent():GetColor())
```

Comment:
The bar should not know about the parent. 

Instead I think you need to introduce a "UpdateForegroundColor()" in the bar prototype table, which uses the foreground + foreground gradient + gradient settings to apply the gradient.

And call that function whenever any of those properties change. 

That's a better way to manage this instead of calling the aurabar's Color function.